File: ChickInRun/Assets/SubModel/Weasel/ObjectIns.js
#pragma strict

function Start(){
}
/*
function OnTriggerEnter (col : Collider){

	if(col.gameObject.tag == "Chick"){
		col.GetComponent(PlayerInst).crashVar++;
		col.gameObject.SendMessage("Split");
	}
}
*/

function OnCollisionEnter (col : Collision){

	if(col.gameObject.tag == "Chick"){
		col.gameObject.GetComponent(PlayerInst).crashVar++;
		col.gameObject.SendMessage("Split");
	}
}

File: ChickInRun/Assets/Script/ObjectIns.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class ObjectIns : MonoBehaviour
{

    public GameObject weaselParticle;

    // Start is called before the first frame update
    void Start()
    {
        
    }

    private void OnCollisionEnter(Collision collision)
    {
        if(collision.gameObject.tag == "Chick")
        {
            collision.gameObject.SendMessage("Split");
            Instantiate(weaselParticle, transform.position, transform.rotation);
            Destroy(gameObject);
        }
    }

    // Update is called once per frame
    void Update()
    {
        
    }
}
